packages/backend/src/models/table-metadata.ts

```diff
@@ -40,6 +40,19 @@ class TableMetadata extends Base {
       },
     },
   })
+
+  async validateRowKeys(data: Record<string, unknown>): Promise<boolean> {
+    const columns = await this.$relatedQuery('columns')
```

Comment:
nit: can limit columns to save some compute. 

```suggestion
    const columns = await this.$relatedQuery('columns').select('id')
```